extension/beatsauthextension/authenticator_test.go

```diff
@@ -0,0 +1,163 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package beatsauthextension
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tj/assert"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/configauth"
+	"go.opentelemetry.io/collector/config/confighttp"
+
+	"github.com/elastic/elastic-agent-libs/transport/tlscommon"
+	"github.com/elastic/opentelemetry-collector-components/extension/beatsauthextension/internal/metadata"
+)
+
+// It tests whether VerifyConnection is set on tls.Config
+func TestVerifyConnection(t *testing.T) {
+	testCerts := tlscommon.GenTestCerts(t)
+	fingerprint := tlscommon.GetCertFingerprint(testCerts["ca"])
+	settings := componenttest.NewNopTelemetrySettings()
+
+	httpClientConfig := confighttp.NewDefaultClientConfig()
+	httpClientConfig.Auth = &configauth.Authentication{
+		AuthenticatorID: component.NewID(metadata.Type),
+	}
+
+	testcases := map[string]struct {
+		verificationMode     tlscommon.TLSVerificationMode
+		peerCerts            []*x509.Certificate
+		serverName           string
+		expectedCallback     bool
+		expectingError       bool
+		CATrustedFingerprint string
+		CASHA256             []string
+	}{
+		"CATrustedFingerprint and verification mode:VerifyFull": {
+			verificationMode:     tlscommon.VerifyFull,
+			peerCerts:            []*x509.Certificate{testCerts["correct"], testCerts["ca"]},
+			serverName:           "localhost",
+			expectedCallback:     true,
+			CATrustedFingerprint: fingerprint,
+		},
+		"CATrustedFingerprint and verification mode:VerifyCertificate": {
+			verificationMode:     tlscommon.VerifyCertificate,
+			peerCerts:            []*x509.Certificate{testCerts["correct"], testCerts["ca"]},
+			serverName:           "localhost",
+			expectedCallback:     true,
+			CATrustedFingerprint: fingerprint,
+		},
+		"CATrustedFingerprint and verification mode:VerifyStrict": {
+			verificationMode:     tlscommon.VerifyStrict,
+			peerCerts:            []*x509.Certificate{testCerts["correct"], testCerts["ca"]},
+			serverName:           "localhost",
+			expectedCallback:     true,
+			CATrustedFingerprint: fingerprint,
+			CASHA256:             []string{tlscommon.Fingerprint(testCerts["correct"])},
+		},
+		"CATrustedFingerprint and verification mode:VerifyNone": {
+			verificationMode: tlscommon.VerifyNone,
+			peerCerts:        []*x509.Certificate{testCerts["correct"], testCerts["ca"]},
+			serverName:       "localhost",
```

Comment:
Maybe set this to something other than "localhost", since it shouldn't be checked by VerifyNone